### PR TITLE
Use infinitime_fonts target if fonts/CMakeLists.txt exists

### PR DESCRIPTION
**Pull request by NeroBurner**
Since https://github.com/InfiniTimeOrg/InfiniTime/pull/1097 the fonts
are generated. Support this new way of using fonts.

### CODE REVIEW

**Comment by Riksu9000** [comment]
This seems to fail when using an external InfiniTime repo, because it's not a subdirectory of InfiniSim.
```
CMake Error at CMakeLists.txt:244 (add_subdirectory):
  add_subdirectory not given a binary directory but the given source
  directory "/home/riku/src/InfiniTime/src/displayapp/fonts" is not a
  subdirectory of "/home/riku/src/InfiniSim".  When specifying an out-of-tree
  source a binary directory must be explicitly specified.
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -238,6 +238,13 @@ set_target_properties(QCBOR PROPERTIES LINKER_LANGUAGE C)
 #        )
 target_link_libraries(infinisim PRIVATE QCBOR)
 
+if(EXISTS ${InfiniTime_DIR}/src/displayapp/fonts/CMakeLists.txt)
+  # available since https://github.com/InfiniTimeOrg/InfiniTime/pull/1097
+  message(STATUS "add subdirectory ${InfiniTime_DIR}/src/displayapp/fonts for 'infinitime_fonts' target")
+  add_subdirectory(${InfiniTime_DIR}/src/displayapp/fonts)
```
Comment (NeroBurner) [inline CMakeLists.txt:244]: @Riksu9000 thanks for testing with an external InfiniTime repo. Could you try this modified `add_subdirectory` command. Currently don't have access to my PC

```suggestion
  add_subdirectory(${InfiniTime_DIR}/src/displayapp/fonts fonts)
```

Comment (Riksu9000) [reply CMakeLists.txt:244]: That seems to work 👍

**Review by yehoshuapw — APPROVED** [review]
I didn't check to make sure it works, since it seems that has been done already - but looks good.
thanks!

**Review by JF002 — APPROVED** [review]
It works for me, both on the "older" develop branch (when the fonts were not generated by cmake) and with the new one (with font generated by cmake)!

**Pull request merged**